### PR TITLE
Add 'parallel' and 'execution-group' to actions and wire things up

### PR DESCRIPTION
**Pull request by wallyworld**
Actions gain 2 new attributes:
- parallel (bool): whether to acquire machine lock
- execution-group (string): actions in the same group share the same lock

Actions by default are still serialised on the lock, but charms can now specify actions to run in parallel.
Juju exec scripts run in parallel by default (CLI not yet updated to specify otherwise but will be).

The unit agent runner doesn't yet act on these new attributes - they are just wired up for now.

Brings in updates to these repos to handle the new action metadata:
- juju/charms/v9
- juju/description/v2

## QA steps

Double check `juju exec` and `juju run` haven't regressed.

### CODE REVIEW

**Comment by wallyworld** [comment]
[weave through new action metadata](https://trello.com/c/5XqFyE2n/1335-weave-through-new-action-metadata)

File: worker/uniter/runner/runner.go [hunk 1]
```diff
@@ -312,6 +312,7 @@ func (runner *runner) runJujuRunAction() (err error) {
 		return errors.Trace(err)
 	}
 
+	// TODO(wallyworld) - use parallel and execution group
```
Comment (hpidcock) [inline worker/uniter/runner/runner.go:315]: Is this coming in another PR?

**Review by hpidcock — APPROVED** [review]

File: worker/uniter/runner/runner.go (hunk 1 above)
Comment (wallyworld) [reply worker/uniter/runner/runner.go:315]: yup, next PR

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**